x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +75,14 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        super.postAnalysisVerification(failures);
+        if (right() instanceof EsRelation esr) {
+            if (esr.concreteIndices().contains(esr.indexPattern()) == false) {
```

Comment:
nit: can we add a comment here that this check may be crucial for security, as we cannot deal with aliases adequately, yet?